edx_event_bus_kafka/internal/consumer.py

```diff
@@ -133,6 +134,20 @@ def consume_indefinitely(self):
         """
         Consume events from a topic in an infinite loop.
         """
+
+        def reset_offsets(consumer, partitions):
+            # This is a callback method used on consumer assignment to handle offset reset logic.
+            # We do not want to attempt to change offsets if the offsets is None or if it's an empty list.
+            if not self.offsets:
+                return
+            if len(self.offsets) != len(partitions):
+                logger.error("Offsets given do not match the number of partitions. Skipping reset step.")
+                return
+            for partition, offset in zip(partitions, self.offsets):
```

Comment:
I don't think we're guaranteed which partitions we'll be assigned, or in what order.

Reply:
yeah, this code was very much of the 'let's see what it would look like to pass in a list of offsets', but since we're not going this route, i don't think we have to worry about it.